### PR TITLE
New Ranking, 100% broken

### PR DESCRIPTION
**Pull request by MatejMecka**

### CODE REVIEW

File: settings/default.py [hunk 1]
```diff
@@ -27,6 +27,29 @@
...
-REACTION = "1"
-POSTED_MESSAGE = "2"
+# Event Types
+MESSAGE_REPLY = env.int("MESSAGE_REPLY_TYPE_CODE", 0)
+REACTION = env.int("MESSAGE_REACTION_TYPE_CODE", 1)
+POSTED_MESSAGE = env.int("MESSAGE_POSTED_TYPE_CODE", 2)
+
+MESSAGE_REPLY_POINTS = env.int("MESSAGE_REPLY_TYPE_CODE", 2)
+REACTION_POINTS = env.int("MESSAGE_REACTION_TYPE_CODE", 1)
+POSTED_MESSAGE_POINTS = env.int("MESSAGE_POSTED_TYPE_CODE", 1)
```
Comment (Smephite) [inline settings/default.py:37]: Name of the env variables is duplicate here which should probably not be the case.
Why do we need the type code exposed as env variables again? Can't we hard code them?

Comment (MatejMecka) [reply settings/default.py:37]: Good catch! I'll fix it. As for the moment, the reason why we made them environment variables was to easily experiment with how we prioritize different upvote types and what weights should we give them, without having to push to GitHub all the time

Comment (Smephite) [reply settings/default.py:37]: But we could hardcore the type but at the same time leaving the weights as an env variable

Comment (MatejMecka) [reply settings/default.py]: Done!

File: handlers/account_missing_warning.py [hunk 2]
```diff
@@ -21,15 +20,14 @@ async def fetch_users_missing_pub(conn=conn):
...
-    leaderboard_rows = fetchLeaderboard(conn, fetch_last_tx(), datetime.now())
+    leaderboard_rows = getUpvoteMap(conn, fetch_last_tx())
     user_rows = fetchUserPubKeys(conn)
     missingUsers = []
 
     for row in leaderboard_rows:
-
-        if not [user for user in user_rows if user[0] == row[0]] and row[0] != client.user.id:
-            print(f"{row[0]} has no pub key connected to their account! They are missing out on {row[1]} upvotes.")
+        if not [user for user in user_rows if user[0] == row] and row != client.user.id:
```
Comment (Kanaye) [inline handlers/account_missing_warning.py:28]: This syntax still looks *insane* to me ... but uhh .... I guess that's just 🐍 stuff 🤣

File: helpers/database.py [hunk 3]
```diff
@@ -232,11 +242,47 @@ def fetchLeaderboard(conn: Connection, dateFrom: datetime = None, dateTo: dateti
...
+    )
+    return c.fetchall()
+
+
+def fetchMessages(
+    conn: Connection, minValue: int, maxValue: int, vote_type: int, dateFrom: datetime = None, dateTo: datetime = None
+):
+    """
+    Return the Number of total chars.
+    Love you Kanaye for the SQL Query <3 Thank you :)
```
Comment (Kanaye) [inline helpers/database.py:263]: ❤️

File: helpers/database.py [hunk 4]
```diff
@@ -232,11 +242,47 @@ def fetchLeaderboard(conn: Connection, dateFrom: datetime = None, dateTo: dateti
...
+
+    c = conn.cursor()
+    c.execute(
+        prepareQuery(
+            """
+            SELECT user_id, vote_type, COUNT(messages.character_count) as votes FROM votes_history
+            INNER JOIN messages ON messages.message_id = votes_history.message_id
+            WHERE vote_type = 2
+            AND messages.character_count >= ? -- you will need to replace this
+            AND messages.character_count < ? -- and this
```
Comment (Kanaye) [inline helpers/database.py:278]: Did you keep the "comments" on purpose ? 🤣

**Review by Kanaye — APPROVED** [review]
Up to you if you keep the "comments" or not, everything else looks 🐍y to me 😛

File: helpers/database.py (hunk 4 above)
Comment (MatejMecka) [reply helpers/database.py:278]: Yes 🤣 
I wanted to preserve the SQL query the way you have sent it to me

File: helpers/database.py (hunk 3 above)
Comment (MatejMecka) [reply helpers/database.py:263]: 💙

File: handlers/account_missing_warning.py (hunk 2 above)
Comment (MatejMecka) [reply handlers/account_missing_warning.py:28]: Python is fun! The fun is in making the code look insane but readable 🤣

**Pull request merged**